src/transform/transform.ts

```diff
@@ -94,7 +94,7 @@ function schemaTypeToGoType(codeModel: CodeModel, schema: Schema, inBody: boolea
       const dictSchema = <DictionarySchema>schema;
       const dictElem = <Schema>dictSchema.elementType;
       dictElem.language.go!.name = schemaTypeToGoType(codeModel, dictElem, inBody);
-      return `map[string]*${dictElem.language.go!.name}`;
+      return `map[string]${dictElem.language.go!.name}`;
```

Comment:
Doesn't this mean that we should check for the x-ms-nullable field here and add the * depending on its existence?

Reply:
Yes, unfortunately it needs some work in autorest.

Reply:
Here's the issue tracking support for this on our side.  https://github.com/Azure/autorest.go/issues/340